examples/10.audit/README.md

```diff
@@ -0,0 +1,313 @@
+    some permitEncrypt, permitDecrypt, shift
+
+    # Extract the values.  Each of these will fail if the value is not as expected.
+    # permitEncrypt is a (single-byte) boolean value at offset 0.
+    permitEncrypt = value_as_boolean(integer_from_hex_string(capability.contents, 0, 1))
+
+    # permitDecrypt is a (single-byte) boolean value at offset 1.
+    permitDecrypt = value_as_boolean(integer_from_hex_string(capability.contents, 1, 1))
+
+    # shift is a single-byte integer value at offset 2.
+    shift = integer_from_hex_string(capability.contents, 2, 1)
+    decoded = {
+        "permitEncrypt": permitEncrypt,
+        "permitDecrypt": permitDecrypt,
+        "shift": shift,
+    }
+}
+```
+
+This starts by depending on the rule that we defined first, which will cause this to fail for anything that isn't a capability of the expected type.
+Next, we define three local variables to hold the three fields that we expect.
+We'll extract each of these using `integer_from_hex_string`, a built-in function provided by `cheriot-audit`.
+This takes a string, an offset, and a length and will decode a little-endian integer from the hex string provided.
+We're extracting three one-byte integers at offsets 0, 1, and 2.
+We're then converting the first two to booleans.
+
+Finally, if all of that worked, we're returning a new object that mirrors the C structure that represents our capability.
+We can use this with another comprehension to extract and decode all valid capabilities.
+This is sufficiently useful that we'll define a new rule for it:
+
+```rego
+all_valid_caesar_capabilities = [{"owner": owner, "capability": decode_user_key_capability(c)} | c = input.compartments[owner].imports[_];         is_sealed_caesar_capability(c)]
+```
+
+This is a slightly more complex comprehension.
+The result is defining a new object with both the owner and the decoded capability.
+Try running this:
+
+```rego
+data.caesar.all_valid_caesar_capabilities
+```
+
+You should see something like this:
+
+```rego
+[
+  {
+    "capability": {
+      "permitDecrypt": true,
+      "permitEncrypt": false,
+      "shift": 95
+    },
+    "owner": "consumer"
+  },
+  {
+    "capability": {
+      "permitDecrypt": false,
+      "permitEncrypt": true,
+      "shift": 95
+    },
+    "owner": "producer"
+  }
+]
+```
+
+Check these values with the ones declared in the source code (in [`producer.cc`](producer.cc) and [`consumer.cc`](consumer.cc)).
+
+Defining our requirements
+-------------------------
+
+Now that we have all of the helpers that let us inspect the linked image, let's define a `valid` rule that defines the policy for our linked compartment.
+We'll start here by depending on the `valid` rule from the RTOS itself:
+
+```rego
+    data.rtos.valid
+```
+
+This performs some sanity checking on the RTOS core, such as ensuring that only the allocator can read hazard-pointer slots, that allocator capabilities are all valid, and so on.
```

Comment:
Do you want to link to it?  (https://github.com/CHERIoT-Platform/cheriot-audit/blob/main/rtos.hh , I assume?)